Sources/ComposableArchitecture/SharedState/PersistenceKey.swift

```diff
@@ -11,32 +11,35 @@
   /// <doc:SharingState#Custom-persistence> section.
   public protocol PersistenceKey<Value>: Hashable {
     associatedtype Value
-    associatedtype Updates: AsyncSequence = _Empty<Value?> where Updates.Element == Value?
 
     /// Loads the freshest value from storage. Returns `nil` if there is no value in storage.
     func load() -> Value?  // TODO: Should this be throwing?
     
     /// Saves a value to storage.
     func save(_ value: Value)
 
-    /// An async sequence that emits when the external storage is changed from the outisde.
-    ///
-    /// An "empty" sequence that does not emit and finishes immediately is provided by default.
-    var updates: Updates { get }
+    /// Subscribes to external updates.
+    func subscribe(didSet: @escaping (Value?) -> Void) -> Shared<Value>.Subscription
   }
 
-  extension PersistenceKey where Updates == _Empty<Value?> {
-    public var updates: _Empty<Value?> {
-      _Empty()
+  extension PersistenceKey {
+    public func subscribe(didSet: @escaping (Value?) -> Void) -> Shared<Value>.Subscription {
+      Shared.Subscription {}
     }
   }
 
-  public struct _Empty<Element>: AsyncSequence {
-    public struct AsyncIterator: AsyncIteratorProtocol {
-      public func next() async throws -> Element? { nil }
-    }
-    public func makeAsyncIterator() -> AsyncIterator {
-      AsyncIterator()
+  extension Shared {
+    public class Subscription {
+      let onCancel: () -> Void
+      init(onCancel: @escaping () -> Void) {
```

Comment:
```suggestion
      public init(onCancel: @escaping () -> Void) {
```

Should this initializer be public since custom `PersistenceKey`s need to be able to return this from `subscribe(didSet:)`?

Reply:
Yup, I think so, thanks!